components/filter_adult/src/lib.rs

```diff
@@ -0,0 +1,105 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+mod error;
+
+pub use error::{ApiError, ApiResult, Result};
+use md5::{Digest, Md5};
+use std::collections::HashSet;
+
+uniffi::setup_scaffolding!("filter_adult");
+
+#[derive(uniffi::Object)]
+pub struct FilterAdultComponent {
+    inner: FilterAdultInner,
+}
+
+#[uniffi::export]
+impl FilterAdultComponent {
+    /// Construct a new FilterAdultComponent
+    #[uniffi::constructor]
+    pub fn new() -> ApiResult<Self> {
+        Ok(Self {
+            inner: FilterAdultInner::new().unwrap(),
+        })
+    }
+
+    /// Check if a URL is in the adult domain list
+    pub fn contains(&self, base_domain_to_check: &str) -> ApiResult<bool> {
+        Ok(self.inner.contains(base_domain_to_check))
+    }
+}
+
+struct FilterAdultInner {
+    hashes: HashSet<[u8; 16]>,
+}
+
+impl FilterAdultInner {
+    pub fn new() -> Result<Self> {
+        let adult_set_bytes = include_bytes!("adult_set.bin");
+        let mut hashes = HashSet::new();
+        for chunk in adult_set_bytes.chunks_exact(16) {
+            let mut arr = [0; 16];
+            arr.copy_from_slice(chunk);
```

Comment:
I think this is needed to convert the reference to an owned array.  You might be able to get away without it by using `hashes: HashSet<&'static [u8; 16]>`.  Either way is fine with me though.